ryhope/src/storage/pgsql/storages.rs

```diff
@@ -125,6 +125,30 @@ where
         }
     }
 
+    /// Retrieve all the (key, payload) pairs valid at a given epoch
+    fn fetch_all_pairs(
+        db: DBPool,
+        table: &str,
+        epoch: Epoch,
+    ) -> impl Future<Output = Result<HashMap<Self::Key, V>>> + std::marker::Send {
```

Comment:
nitpick but if you allow to give a vector of epochs as input then the returned structure is exactly the same as the second field of widelineage. Sounds like redundant structuring here.

Reply:
Is there a case where we want to fetch all values for multiple epochs?